lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,278 @@
+		if err != nil {
+			ht.Fatalf("unable to send '%v' coins to %v: %v", amt, target.Name(), err)
+		}
+	}
+
+	// Helper function to sweep funds from a node wallet.
+	var minerAddr btcutil.Address
+	sweepNodeWalletAndAssert := func(node *lntest.HarnessNode) {
+		// Send funds back to the miner node.
+		if minerAddr == nil {
+			var err error
+			minerAddr, err = net.Miner.NewAddress()
+			if err != nil {
+				ht.Fatalf("unable to create new miner address: %v", err)
+			}
+		}
+
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		if _, err := carol.SendCoins(ctxt, &lnrpc.SendCoinsRequest{
+			Addr:    minerAddr.String(),
+			SendAll: true,
+		}); err != nil {
+			ht.Fatalf("unable to sweep coins from node %v: %v", node.Name(), err)
+		}
+
+		// Ensures we don't leave any transaction in the mempool after sweeping.
+		mineBlocks(ht, net, 1, 1)
+
+		// Make sure all funds were swept.
+		checkChannelBalance(node, 0)
+	}
+
+	var testCases = []chanFundMaxTestCase{
+		{
+			name:               "wallet amount is dust",
+			amount:             2000,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount < min chan size (~18000sat)",
+			amount: 18000,
+			// Using a feeRate of 1 sat/bytes ensures that we test for
+			// min chan size and not excessive fees.
+			feeRate:            1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount > min chan size (37000sat)",
+			amount: 37000,
+			// The transaction fee to open the channel must be subtracted from
+			// Carol's balance (since wallet balance < max-chan-size).
+			carolBalance: btcutil.Amount(37000) - fundingFee(1, false),
+		},
+
+		{
+			name:         "wallet amount > max chan size (20000000sat)",
+			amount:       20000000,
+			carolBalance: lnd.MaxBtcFundingAmount,
+		},
+
+		// This test is actually not for `fundmax` specifically but opening
+		// channels in general.
+		//! triggers error in `rpcserver.go` as required localFundingAmt becomes
+		//! greater than MaxFundingAmount
+		{
+			name:               "wallet amount > max chan size, push amount == max-chan-size",
+			amount:             20000000,
+			pushAmt:            lnd.MaxBtcFundingAmount,
+			chanOpenShouldFail: true,
+		},
+
+		// This test is actually not for `fundmax` specifically but opening
+		// channels in general.
+		//! triggers ErrFunderBalanceDust in `lnwallet/reservation.go`
+		{
+			name:               "wallet amount > max chan size, push amount == max-chan-size - 1",
+			amount:             20000000,
+			pushAmt:            lnd.MaxBtcFundingAmount - 1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:         "wallet amount > max chan size, push amount 16766000",
+			amount:       20000000,
+			pushAmt:      16766000,
+			carolBalance: lnd.MaxBtcFundingAmount - 16766000,
+		},
+	}
+
+	for _, testCase := range testCases {
+		success := ht.t.Run(testCase.name, func(t *testing.T) {
+
+			sendCoinsAndAssert(testCase.amount, carol)
+			defer func() {
```

Comment:
could make this a helper function as well that accepts the node and test case as parameter.